### DISCUSSION TITLE
NequIP with water data

### DISCUSSION THREAD

**Discussion by nisargjoshi10** [mir-group/nequip#203] [Q&A]
Hey all,

I am trying to develop a potential with water data. I have a dataset for 192 atoms water system from AIMD simulation in VASP at 330K temperature. I'm training 1000 frames and using 200 for validation. I am attaching the loss plots and my training script.

After training, I am using the potential to run a small simulation in lammps and observing temperature fluctuations at the end and also observing the radial distribution function plot. I am in the process of understanding the workflow, since I want to run a bunch of simulations with the trained potential. If you have any comments or suggestions please let me know and I can send any other required data to get optimum results.

Thanks for the help!
[water_train.txt](https://github.com/mir-group/nequip/files/8516991/water_train.txt)
![water-vasp-sim5-run1_tr](https://user-images.githubusercontent.com/56214118/164116974-b9fb2b60-3a5f-4788-a37e-fae9732ac2f4.png)
![water-vasp-sim5-run1_v](https://user-images.githubusercontent.com/56214118/164116985-4e1ea105-85f0-4576-b031-bba77bab1e03.png)

Simulation script
```
# bulk water
  
units           real
boundary        p p p
newton          off
atom_style      atomic

neighbor        2.0 bin
neigh_modify    delay 0 every 1 check yes

read_data       water.lmp
mass            1 16
mass            2 2

variable        sysvol          equal vol
variable        sysmass         equal mass(all)/6.0221367e+23
variable        sysdensity      equal v_sysmass/v_sysvol/1.0e-24

pair_style      nequip
pair_coeff      * * water-vasp-sim5-run1.pth O H

#minimization
thermo_style    custom step pe ke etotal temp press vol
thermo          100
min_style cg
minimize 1e-6 1e-6 5000 10000

reset_timestep  0
velocity        all create 330.0 23456789

fix             1 all nvt temp 330.0 330.0 5
timestep        0.05
thermo_style    custom step pe ke etotal v_sysdensity temp press vol
thermo          100
dump            1 all custom 100 water.dump id type x y z

run             3000
```

Simulation results

```
Per MPI rank memory allocation (min/avg/max) = 4.371 | 4.371 | 4.371 Mbytes
Step PotEng KinEng TotEng v_sysdensity Temp Press Volume
       0    -977.7583    187.88046   -789.87784    1.1028236          330     4456.164    1927.3176
     100   -965.72546    178.00524   -787.72022    1.1028236    312.65481    4221.9428    1927.3176
     200    -938.7674    180.14281   -758.62459    1.1028236    316.40931    4272.6418    1927.3176
     300   -924.46008    184.02431   -740.43578    1.1028236    323.22692    4364.7035    1927.3176
     400   -920.32599    192.38895   -727.93704    1.1028236    337.91887    4563.0967    1927.3176
     500   -907.42053    186.46285   -720.95768    1.1028236    327.51006    4422.5411    1927.3176
     600   -887.56122    182.37778   -705.18344    1.1028236     320.3349    4325.6511    1927.3176
     700   -867.38782    187.03591    -680.3519    1.1028236    328.51661     4436.133    1927.3176
     800   -852.56091    181.85492   -670.70599    1.1028236    319.41653    4313.2498    1927.3176
     900   -844.97577    187.41392   -657.56185    1.1028236    329.18056    4445.0987    1927.3176
    1000   -842.32452    195.32103    -647.0035    1.1028236    343.06889    4632.6401    1927.3176
    1100   -842.36249    190.32646   -652.03603    1.1028236    334.29625    4514.1785    1927.3176
    1200   -844.87799    163.09731   -681.78068    1.1028236    286.46999    3868.3553    1927.3176
    1300     -855.224    181.35199   -673.87201    1.1028236    318.53316    4301.3212    1927.3176
    1400    -861.1239    200.34663   -660.77727    1.1028236    351.89604    4751.8377    1927.3176
    1500   -864.08667    193.32888   -670.75779    1.1028236    339.56981    4585.3902    1927.3176
    1600   -859.25836    178.66337     -680.595    1.1028236    313.81077    4237.5523    1927.3176
    1700    -856.8501    182.03346   -674.81664    1.1028236    319.73012    4317.4844    1927.3176
    1800   -861.77112    194.52884   -667.24228    1.1028236    341.67746    4613.8509    1927.3176
    1900   -863.22626    191.20442   -672.02184    1.1028236    335.83832     4535.002    1927.3176
    2000   -862.54633    169.70998   -692.83635    1.1028236    298.08471     4025.195    1927.3176
    2100   -865.31335    165.59562   -699.71773    1.1028236    290.85811    3927.6105    1927.3176
    2200   -868.00684     201.3788   -666.62804    1.1028236    353.70898    4776.3188    1927.3176
    2300    -864.9668    194.72092   -670.24587    1.1028236    342.01484    4618.4067    1927.3176
    2400   -856.57605    174.28899   -682.28706    1.1028236    306.12746    4133.8005    1927.3176
    2500   -859.60034    188.22115   -671.37919    1.1028236     330.5984    4464.2445    1927.3176
    2600   -868.13434    213.69258   -654.44176    1.1028236    375.33735    5068.3781    1927.3176
    2700   -910.24957    224.17515   -686.07443    1.1028236     393.7493    5317.0044    1927.3176
    2800   -937.94049    274.80548   -663.13501    1.1028236    482.67823    6517.8587    1927.3176
    2900   -855.49115    138263.04    137407.55    1.1028236    242850.19    3279334.1    1927.3176
    3000   -844.73212    1783.8858     939.1537    1.1028236    3133.2813    42310.349    1927.3176
Loop time of 3563.2 on 1 procs for 3000 steps with 192 atoms
```

**Comment by Linux-cpp-lisp** [comment]
@nisargjoshi10 what is your question?

**Reply by simonbatzner** [reply]
Your units are real, that means your NequIP model should have been trained on data in kcal/mol/A is that indeed the case? You also use a thermostat interaction time of 5fs (that's quite aggressive). 

Can you share what units these errors are in, otherwise it's hard to judge?

**Reply by nisargjoshi10** [reply]
@Linux-cpp-lisp my question was about understanding the temperature fluctuations in the simulation at the end. from @simonbatzner suggestions, I realized my data was in eV and I was using the wrong units. I ran another simulation and seems like I get close to the target temperature at the end.

```
Step PotEng KinEng TotEng v_sysdensity Temp Press Volume
       0    -977.7583    8.1472669   -969.61103    1.1028236          330    4515.2079    1927.3176
     100   -973.27045    3.6948527   -969.57559    1.1028236    149.65772     2047.684    1927.3176
     200   -973.61414    4.0927953   -969.52134    1.1028236    165.77614    2268.2234    1927.3176
     300    -973.7995    4.3752617   -969.42424    1.1028236    177.21727     2424.766    1927.3176
     400   -973.61523    4.3232942   -969.29194    1.1028236    175.11235    2395.9657    1927.3176
     500   -973.52612    4.3767709   -969.14935    1.1028236    177.27839    2425.6024    1927.3176
     600   -973.45398    4.4396877   -969.01429    1.1028236     179.8268    2460.4709    1927.3176
     700   -973.47125    4.5838546    -968.8874    1.1028236    185.66619     2540.368    1927.3176
     800   -973.05646    4.2858589    -968.7706    1.1028236    173.59606    2375.2191    1927.3176
     900   -972.84314    4.1943726   -968.64877    1.1028236    169.89046    2324.5174    1927.3176
    1000   -973.16785    4.6475951   -968.52025    1.1028236    188.24796    2575.6929    1927.3176
    1100   -973.02173    4.6463283    -968.3754    1.1028236    188.19665    2574.9909    1927.3176
    1200   -972.84033    4.6179821   -968.22235    1.1028236     187.0485    2559.2814    1927.3176
    1300   -972.96356    4.9018277   -968.06173    1.1028236    198.54549    2716.5884    1927.3176
    1400   -973.14276    5.2522546   -967.89051    1.1028236    212.73932    2910.7947    1927.3176
    1500   -972.46063    4.7470929   -967.71354    1.1028236    192.27806    2630.8345    1927.3176
    1600   -972.67816    5.1387152   -967.53945    1.1028236    208.14048    2847.8713    1927.3176
    1700   -972.59186    5.2402325   -967.35163    1.1028236    212.25237    2904.1321    1927.3176
    1800   -972.58557    5.4241352   -967.16144    1.1028236    219.70124    3006.0508    1927.3176
    1900   -972.67841    5.7103028    -966.9681    1.1028236    231.29228    3164.6446    1927.3176
    2000   -972.53369    5.7583177   -966.77537    1.1028236    233.23709    3191.2544    1927.3176
    2100   -972.14691      5.57179   -966.57512    1.1028236     225.6819    3087.8809    1927.3176
    2200   -971.75049    5.3740129   -966.37648    1.1028236    217.67106     2978.273    1927.3176
    2300   -971.84937    5.6773457   -966.17202    1.1028236    229.95737    3146.3798    1927.3176
    2400   -971.46381    5.4914659   -965.97234    1.1028236    222.42843    3043.3654    1927.3176
    2500   -971.80481    6.0385755   -965.76623    1.1028236    244.58876     3346.573    1927.3176
    2600    -971.5014     5.952328   -965.54908    1.1028236    241.09536    3298.7747    1927.3176
    2700   -971.44562    6.0987942   -965.34682    1.1028236    247.02788    3379.9462    1927.3176
    2800   -971.86713     6.738724    -965.1284    1.1028236    272.94784    3734.5946    1927.3176
    2900   -971.36774    6.4533658   -964.91437    1.1028236    261.38958    3576.4494    1927.3176
    3000   -970.97748    6.2663733    -964.7111    1.1028236    253.81557    3472.8183    1927.3176
    3100   -970.54291    6.0380487   -964.50486    1.1028236    244.56742     3346.281    1927.3176
    3200   -971.33496    7.0425393   -964.29242    1.1028236    285.25369    3902.9688    1927.3176
    3300   -970.81732    6.7464203    -964.0709    1.1028236    273.25958      3738.86    1927.3176
    3400    -971.0567    7.2004624   -963.85624    1.1028236    291.65027    3990.4897    1927.3176
    3500   -970.46765    6.8333744   -963.63428    1.1028236     276.7816    3787.0498    1927.3176
    3600   -970.14294    6.7193726   -963.42357    1.1028236    272.16402    3723.8701    1927.3176
    3700   -970.60608    7.3870494   -963.21903    1.1028236    299.20786     4093.896    1927.3176
    3800   -970.37695    7.3718434   -963.00511    1.1028236    298.59195    4085.4689    1927.3176
    3900   -970.06177     7.254138   -962.80763    1.1028236    293.82437    4020.2366    1927.3176
    4000   -969.94861      7.34694   -962.60167    1.1028236    297.58325    4071.6674    1927.3176
    4100   -970.12292    7.7153342   -962.40759    1.1028236    312.50483    4275.8311    1927.3176
    4200   -969.77338    7.5672668   -962.20611    1.1028236    306.50745    4193.7723    1927.3176
    4300   -969.79419    7.7840654   -962.01012    1.1028236    315.28875    4313.9219    1927.3176
    4400    -969.4267    7.6124223   -961.81427    1.1028236    308.33645    4218.7974    1927.3176
    4500   -969.15997    7.5352304   -961.62474    1.1028236    305.20984    4176.0178    1927.3176
    4600   -969.73334    8.2933039   -961.44003    1.1028236    335.91513    4596.1414    1927.3176
    4700   -969.36292     8.111248   -961.25167    1.1028236    328.54108    4495.2462    1927.3176
    4800   -969.43024    8.3429956   -961.08724    1.1028236    337.92787    4623.6805    1927.3176
    4900   -969.06293    8.1413318    -960.9216    1.1028236     329.7596    4511.9186    1927.3176
    5000   -969.07092    8.3017279   -960.76919    1.1028236    336.25635      4600.81    1927.3176
```
also @simonbatzner did you mean to change the damping parameter in your comment earlier?

**Reply by simonbatzner** [reply]
What do you mean? I was referring to the real units you set, which gives you fs so you set a damping parameter of 5fs.

**Reply by nisargjoshi10** [reply]
I was referring to the units used earlier. My training data had units in `eV` and I realized reals uses kcal/mol which might be causing any issues. I changed the units later to metal for my simulation and I'm getting good results. 

For the damping parameter, I was trying to follow 100*dt and didnt realize 5fs would be too extreme. Anyways, changing that did help me with my results! Working on training more potentials and exploring them.

Thanks to both of you for quick responses and creating this awesome tool! (I'm really excited to use this for my research)

**Reply by simonbatzner** [reply]
Keep in mind that when you switch the units from real to metal (as you should) the time units also changes from fs to ps. You previously set a time step of 0.05fs (which is quite short, I think 0.5 fs should be fine for this system). So now you'll have to convert that to ps, so I'd set something like `timestep 0.0005` to get 0.5fs.

**Reply by simonbatzner** [reply]
https://docs.lammps.org/units.html

**Reply by nisargjoshi10** [reply]
you're right! thanks for the headsup!
